### PR TITLE
Improved handling of connection

### PR DESCRIPTION
**Pull request by am0s**
Misc improvements to the socket connection handling.

* Avoids inifinite connection loops if the client was told to disconnect.
* Allows a caller to listen for connection changes, e.g. to listen for a disconnection.
* The time between connection retries can be specified by the caller.

### CODE REVIEW

File: pychromecast/socket_client.py [hunk 1]
```diff
@@ -809,4 +891,10 @@ def tear_down(self):
         """ Called when controller is destroyed. """
         super(ReceiverController, self).tear_down()
 
+        self.status = None
+        self.launch_failure = None
+        self.app_to_launch = None
```
Comment (balloob) [inline pychromecast/socket_client.py]: These variables seem not to be used.

Comment (balloob) [inline pychromecast/socket_client.py]: Ah, I see now that not all code is included in this PR but some have been committed. Nevermind.

**Comment by balloob** [comment]
It looks good. Awesome.

If you can give me your PyPi username, I will add you as an owner so you can make releases too.

File: pychromecast/socket_client.py (hunk 1 above)
Comment (am0s) [inline pychromecast/socket_client.py]: Ah yes, the previous commits were meant as a pull request as well but I misunderstood the UI in sourcetree and it got pushed directly.

**Pull request merged**